blackboxopt/optimization_loops/utils.py

```diff
@@ -55,7 +56,10 @@ def evaluation_function_wrapper(
     """
     try:
         evaluation = evaluation_function(evaluation_specification)
-    except Exception:
+    except Exception as e:
+        if exit_on_unhandled_exception:
+            raise e
```

Comment:
Just an idea: Could it be useful to have a custom exception, which the user can easier catch and work with? Something like:
```
    except Exception as e:
        if exit_on_unhandled_exception:
            raise BboEvaluationFunctionError from e
```